code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -1042,18 +853,21 @@ var/global/list/obj/item/device/pda/PDAs = list()
 
 
 /obj/item/device/pda/proc/explode() //This needs tuning.
-	if(!src.detonate) return
-	var/turf/T = get_turf(src.loc)
+	if(!detonate) return
+	var/turf/T = get_turf(src)
 
 	if (ismob(loc))
 		var/mob/M = loc
-		M.show_message("<span class='danger'>Your [src] explodes!</span>", 1)
+		M.show_message("<span class='userdanger'>Your [src] explodes!</span>", 1)
+	else
+		visible_message("<span class='danger'>[src] explodes!</span>", "<span class='warning'>You hear a loud *pop*!</span>")
 
 	if(T)
 		T.hotspot_expose(700,125)
-
-		explosion(T, -1, -1, 2, 3)
-
+		if(istype(cartridge, /obj/item/weapon/cartridge/syndicate))
+			explosion(T, -1, 1, 3, 4)

```

Comment:
Wow. That's a part of other PR that ended up here. It was adding a self-destruction option to PDAs with Syndicate Uplink and/or Detomatrix. It's useless without the rest of PR because traitor PDAs are detomatrix-proof and can't explode. I'll just leave that in.